tests/unittests/sources/test_wsl.py

```diff
@@ -460,6 +461,53 @@ def test_get_data_jinja(self, m_lsb_release, paths, tmpdir):
             cast(MIMEMultipart, ud), "text/cloud-config"
         ), "No cloud-config part should exist"
 
+    @pytest.mark.parametrize("with_agent_data", [True, False])
+    @mock.patch("cloudinit.util.lsb_release")
+    def test_get_data_x(
+        self, m_lsb_release, with_agent_data, caplog, paths, tmpdir
+    ):
+        """
+        Assert behavior of empty .cloud-config dir with and without agent data
+        """
+        m_lsb_release.return_value = SAMPLE_LINUX_DISTRO
+        data_path = tmpdir.join(".cloud-init", f"{INSTANCE_NAME}.user-data")
+        data_path.dirpath().mkdir()
+
+        if with_agent_data:
+            ubuntu_pro_tmp = tmpdir.join(".ubuntupro", ".cloud-init")
+            os.makedirs(ubuntu_pro_tmp, exist_ok=True)
```

Comment:
Nit, let's move this dir creation outside of the with_agent_data clause to ensure we are coping with presence of an empty directory as the unit test suggests.